test/integration/targets/elb_target/playbooks/version_fail.yml

```diff
@@ -4,31 +4,38 @@
 
   tasks:
     - name: set up aws connection info
-      set_fact:
-        aws_connection_info: &aws_connection_info
-          aws_access_key: madeup
-          aws_secret_key: madeup
-          security_token: madeup
+      module_defaults:
+        group/aws:
+          aws_access_key: "{{ aws_access_key }}"
+          aws_secret_key: "{{ aws_secret_key }}"
+          security_token: "{{ security_token | default(omit) }}"
           region: "{{ aws_region }}"
-      no_log: yes
+      block:
+      - name: set up testing target group (type=ip)
+        elb_target_group:
+          state: present
+          #name: "{{ resource_shortprefix }}-tg"
+          name: "ansible-test-{{ resource_prefix | regex_search('([0-9]+)$') }}-tg"
+          health_check_port: 80
+          protocol: http
+          port: 80
+          vpc_id: 'vpc-abcd1234'
+          target_type: ip
+          tags:
+            Description: "Created by {{ resource_prefix }}"
+        register: elb_target_group_type_ip
+        ignore_errors: yes
 
-    - name: set up testing target group (type=ip)
-      elb_target_group:
-        name: "ansible-test-{{ resource_prefix | regex_search('([0-9]+)$') }}-tg"
-        health_check_port: 80
-        protocol: http
-        port: 80
-        vpc_id: 'vpc-abcd1234'
-        state: present
-        target_type: ip
-        tags:
-          Description: "Created by {{ resource_prefix }}"
-        <<: *aws_connection_info
-      register: elb_target_group_type_ip
-      ignore_errors: yes
+      - name: check that setting up target group with type=ip fails with friendly message
+        assert:
+          that:
+            - elb_target_group_type_ip is failed
+            - "'msg' in elb_target_group_type_ip"
 
-    - name: check that setting up target group with type=ip fails with friendly message
-      assert:
-        that:
-          - elb_target_group_type_ip is failed
-          - "'msg' in elb_target_group_type_ip"
+      # In the off-chance that this went (partially) through when it shouldn't...
+      always:
+      - name: Remove testing target group (type=ip)
+        elb_target_group:
+          state: absent
```

Comment:
This has actually uncovered a bug in the arg spec (or docs) for this module.  `elbv2.delete_target_group` only needs an ARN, however the module defaults `target_type` to instance and requires `['protocol', 'port', 'vpc_id']` if `target_type` is set.  So this task is failing:
`target_type is instance but all of the following are missing: protocol, port, vpc_id`

@tremble - would you be up for tackling that bug here, or else we'll need to add the extra parameters like in https://github.com/ansible/ansible/blob/devel/test/integration/targets/elb_target/playbooks/roles/elb_target/tasks/main.yml#L343:L351